tmt/base.py

```diff
@@ -2091,6 +2086,15 @@ def __init__(
     def from_tree(cls, tree: 'tmt.Tree') -> List['Story']:
         return tree.stories()
 
+    def _export(self, *, keys: Optional[List[str]] = None) -> tmt.export._RawExportedInstance:
+        data = super()._export(keys=keys)
+
+        for key, value in data.items():
+            if key == 'priority' and value is not None:
+                data[key] = cast(StoryPriority, value).value
```

Comment:
What about this?

```suggestion
                data[key] = str(value)
```

Seems a bit more readable and we have the `__str__` method ready:

https://github.com/teemtee/tmt/blob/8d414522d9dd0988f368de3cb368751c2ae779d1/tmt/base.py#L2037-L2039

Reply:
I'd still go with `.value`, it's a common way how to access enum member properties, `.name` & `.value`, and it's used with other enums in tmt code base (`ResultOutcome`, `LinterOutcome`, `tmt.log.Topic`).

Reply:
Ok, I don't insist on this. Just wonder, what is the use case for `__str__` then if we're not using it when we need to convert the priority to string, which, I thought, is the main use case of having the method :)

Reply:
That `__str__()` feels to me more like a hack, to make the enum play nicely with fmf's `filter()` assumption that everything can be a string, and the default `__str__` of `Enum` used member name rather than value. So in this sense, it's rather tailored for one specific use. For me, using `.name` and `.value` is more explicit and clear about what part of the enum member is needed, without giving the enum class a chance to mess with our code by changing the implementation of `__str__()`. It still looks like various parts of the code have a different idea about how stringified enum members should look in their context, some need its name, some its value.

I don't mind switching to `__str__()` everywhere instead of `.value`, but it must be stated first that when looking for stringified, a printable representation of the string, `str(member)` is the way to do it, and all enums and their users must be aware of this assumption. I would even write such a patch, but I'd rather avoid changing one line for one enum only, that would create an exception harder to track.